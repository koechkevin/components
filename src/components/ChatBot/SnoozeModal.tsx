import { Row } from 'antd';
import { ModalProps } from 'antd/lib/modal';
import React, { FC } from 'react';

import Button from '../Button/Button';
import Modal from '../Modal/Modal';

interface Props extends ModalProps {
  reminderList: any[];
  onSnooze?: (params?: object) => void;
}

const SnoozeModal: FC<Props> = (props) => {
  const { onSnooze, reminderList, ...otherProps } = props;

  return (
    <Modal centered width={500} title="Snooze Reminder" footer={null} {...otherProps}>
      <Row>
        {reminderList && reminderList.length
          ? reminderList.map((item, index: number) => (
              <Button block ghost key={index} style={{ marginBottom: 16 }} onClick={() => onSnooze && onSnooze(item)}>
                {item.label}
              </Button>
            ))
          : null}
      </Row>
    </Modal>
  );
};

export default SnoozeModal;
