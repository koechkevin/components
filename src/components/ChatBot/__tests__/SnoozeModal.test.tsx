import { mount } from 'enzyme';
import React from 'react';

import { Button } from '../../..';
import SnoozeModal from '../SnoozeModal';

describe('Tests for snooze modal', () => {
  const props = {
    reminderList: [
      { label: 'Snooze for 20 minutes', value: 20 * 60 },
      { label: 'Snooze for 1 hour', value: 60 * 60 },
    ],
    onSnooze: jest.fn(),
  };

  const wrapper = mount(
    <div>
      <SnoozeModal visible {...props} />
    </div>,
  );

  it('should render snooze modal without crashing', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the modal title', () => {
    expect(wrapper.contains('Snooze Reminder')).toEqual(true);
  });

  it('should display reminder list options', () => {
    expect(wrapper.contains('Snooze for 20 minutes')).toEqual(true);
    expect(wrapper.contains('Snooze for 1 hour')).toEqual(true);
  });

  it('should call onSnooze when reminder button is clicked', () => {
    const button = wrapper.find(Button).last();
    button.simulate('click');
    expect(button.contains('Snooze for 1 hour')).toEqual(true);
    expect(props.onSnooze).toHaveBeenCalled();
  });
});
