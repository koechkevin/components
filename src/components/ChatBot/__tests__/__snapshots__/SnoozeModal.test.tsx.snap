// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for snooze modal should render snooze modal without crashing 1`] = `ReactWrapper {}`;
